packages/fast-usdc/src/exos/transaction-feed.js

```diff
@@ -73,45 +75,94 @@ export const prepareTransactionFeedKit = (zone, zcf) => {
               seat.exit();
-              return admin.initOperator(operatorId);
+              return creator.initOperator(operatorId);
             },
             INVITATION_MAKERS_DESC,
           );
         },
+        /** @param {string} operatorId */
+        async initOperator(operatorId) {
+          const { operators, pending } = this.state;
+          trace('initOperator', operatorId);
+
+          const operatorKit = makeOperatorKit(
+            operatorId,
+            this.facets.operatorPowers,
+          );
+          operators.init(operatorId, operatorKit);
+          pending.init(operatorId, makeScalarBigMapStore('pending evidence'));
+
+          return operatorKit;
+        },
+
         /** @param {string} operatorId */
         async removeOperator(operatorId) {
-          const { operators: oracles } = this.state;
+          const { operators } = this.state;
           trace('removeOperator', operatorId);
-          const kit = oracles.get(operatorId);
-          kit.admin.disable();
-          oracles.delete(operatorId);
+          const operatorKit = operators.get(operatorId);
+          operatorKit.admin.disable();
+          operators.delete(operatorId);
         },
       },
+      operatorPowers: {
+        /**
+         * Add evidence from an operator.
+         *
+         * @param {CctpTxEvidence } evidence
+         * @param {OperatorKit} operatorKit
+         */
+        submitEvidence(evidence, operatorKit) {
+          const { pending } = this.state;
+          trace(
+            'submitEvidence',
+            operatorKit.operator.getStatus().operatorId,
+            evidence,
+          );
+          const { operatorId } = operatorKit.operator.getStatus();
 
-      admin: {
-        /** @param {string} operatorId */
-        async initOperator(operatorId) {
-          const { operators: oracles } = this.state;
-          trace('initOperator', operatorId);
+          // TODO verify that the operator is one made by this exo
+          // This doesn't work...
+          // operatorKit === operators.get(operatorId) ||
+          //   Fail`operatorKit mismatch`;
 
-          const oracleKit = makeOperatorKit(operatorId);
-          oracles.init(operatorId, oracleKit);
+          // TODO validate that it's valid before accepting
+          const { txHash } = evidence;
 
-          return oracleKit;
-        },
+          // accept the evidence
+          {
+            const pendingStore = pending.get(operatorId);
+            if (pendingStore.has(txHash)) {
+              trace(`operator ${operatorId} already reported ${txHash}`);
+            } else {
+              pendingStore.init(txHash, evidence);
+            }
+          }
+
+          // check agreement
+          const found = [];
+          for (const pendingStore of pending.values()) {
+            if (pendingStore.has(txHash)) {
+              found.push(evidence);
+            }
+          }
+          if (found.length < pending.getSize()) {
```

Comment:
I suppose you considered...

```suggestion
          const found = [...pending.values()].filter(store => store.has(txHash);
          if (found.length < pending.getSize()) {
```

It's a bummer that we have to materialize `[...x.values()]` before calling `.filter()`.

Reply:
thanks. I had gone in another direction and refactored it to this, neglected to try simplifying